tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/XUnit/XUnitTestOutputHelperQueueTestOutputIntegration.cs

```diff
@@ -58,52 +60,97 @@ internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, string
         private class XUnitLogEvent : DatadogLogEvent
         {
             private readonly string _message;
-            private readonly Context? _context;
+            private readonly Span _span;
 
             public XUnitLogEvent(string message, Span span)
             {
                 _message = message;
-                _context = span is null ? null : new Context(span.TraceId, span.SpanId, span.Context.Origin);
+                _span = span;
             }
 
             public override void Format(StringBuilder sb, LogFormatter formatter)
             {
-                formatter.FormatLog<Context?>(
-                    sb,
-                    _context,
-                    DateTime.UtcNow,
-                    _message,
-                    eventId: null,
-                    logLevel: DirectSubmissionLogLevelExtensions.Information,
-                    exception: null,
-                    (JsonTextWriter writer, in Context? state) =>
+                using var writer = LogFormatter.GetJsonWriter(sb);
+
+                // Based on JsonFormatter
+                writer.WriteStartObject();
+
+                writer.WritePropertyName("ddsource", escape: false);
+                writer.WriteValue("xunit");
+
+                if (formatter.Settings.Host is { } formatterHost)
+                {
+                    writer.WritePropertyName("hostname", escape: false);
+                    writer.WriteValue(formatterHost);
+                }
+                else if (HostMetadata.Instance.Hostname is { } hostname)
+                {
+                    writer.WritePropertyName("hostname", escape: false);
+                    writer.WriteValue(hostname);
+                }
```

Comment:
Will this branch ever be hit? 🤔 The settings already defaults to using `HostMetadata.Instance.Hostname` IIRC?

Reply:
true, removed in https://github.com/DataDog/dd-trace-dotnet/pull/3037/commits/17e24d81ff05bc7a5b9e6987ec43680b365bdf06